google/cloud/aiplatform/training_jobs.py

```diff
@@ -281,46 +333,474 @@ def package_and_copy_to_gcs(
+            fraction_split = FractionSplit(
+                training_fraction=training_fraction_split,
+                validation_fraction=validation_fraction_split,
+                test_fraction=test_fraction_split,
+            )
+
+            # create input data config
+            input_data_config = InputDataConfig(
+                fraction_split=fraction_split,
+                dataset_id=dataset.name,
+                gcs_destination=GcsDestination(output_uri_prefix=base_output_dir),
+            )
+
+        # create training pipeline
+        training_pipeline = TrainingPipeline(
+            display_name=self._display_name,
+            training_task_definition=schema.training_job.definition.custom_task,
+            training_task_inputs=json_format.ParseDict(
+                {
+                    "workerPoolSpecs": [worker_pool_spec],
+                    "baseOutputDirectory": {"output_uri_prefix": base_output_dir},
+                },
+                Value(),
+            ),
+            model_to_upload=managed_model,
+            input_data_config=input_data_config,
+        )
+
+        training_pipeline = self.api_client.create_training_pipeline(
+            parent=initializer.global_config.common_location_path(
+                self.project, self.location
+            ),
+            training_pipeline=training_pipeline,
+        )
+
+        self._gca_resource = training_pipeline
+
+        _LOGGER.info("View Training:\n%s" % self._dashboard_uri())
+        _LOGGER.info("Training Output directory:\n%s " % base_output_dir)
+
+        model = self._get_model()
+
+        if model is None:
+            _LOGGER.warn(
+                "Training did not produce a Managed Model returning None. "
+                + self._model_upload_fail_string
+            )
+        return model
+
+    def _sync_gca_resource(self):
+        """Helper method to sync the local gca_source against the service."""
+        self._gca_resource = self.api_client.get_training_pipeline(
+            name=self.resource_name
+        )
+
+    def _block_until_complete(self):
+        """Helper method to block and check on job until complete."""
+
+        # Used these numbers so failures surface fast
+        wait = 5  # start at five seconds
+        max_wait = 60 * 5  # 5 minute wait
+        multiplier = 2  # scale wait by 2 every iteration
+
+        while self.state not in _PIPELINE_COMPLETE_STATES:
+            self._sync_gca_resource()
+            time.sleep(wait)
+            _LOGGER.info(
+                "Training %s current state:\n%s"
+                % (self._gca_resource.name, self._gca_resource.state)
+            )
+            wait = min(wait * multiplier, max_wait)
+
+        self._raise_failure()
+
+        if self._gca_resource.model_to_upload and not self.is_failed:
+            _LOGGER.info(
+                "Model available at %s" % self._gca_resource.model_to_upload.name
+            )
+
+    def _raise_failure(self):
```

Comment:
The service does not gracefully do this, but it would be nice if the library informs the user of what/where intermediate things were created (service resources, GCS objects, etc) so they know what needs to be cleaned up.  This is perhaps useful for both success and failure cases, but seems more important in the case of failure cases.

Reply:
We log all these items during the run.